cpp/locality/NeighborBond.h

```diff
@@ -87,10 +110,58 @@ struct NeighborBond
         return weight < n.weight;
     }
 
+    unsigned int getQueryPointIdx() const
+    {
+        return query_point_idx;
+    }
+
+    void setQueryPointIdx(unsigned int new_qpidx)
```

Comment:
How about a simpler name?
```suggestion
    void setQueryPointIdx(unsigned int idx)
```